odl/util/utility.py

```diff
@@ -258,6 +251,92 @@ def is_complex_floating_dtype(dtype):
     return np.issubsctype(dtype, np.complexfloating)
 
 
+def real_dtype(dtype, use_default=False, default=None):
+    """Return the real counterpart of ``dtype`` if existing.
+
+    Parameters
+    ----------
+    dtype :
+        Real or complex floating point data type. It can be given in any
+        way the `numpy.dtype` constructor understands.
+    use_default : bool, optional
```

Comment:
Why is this needed. Wouldnt `default=None` imply that a error should be raised and `default != None` mean that it should be used as a default?